src/write.jl

```diff
@@ -25,6 +25,11 @@ Write JSON.
 ## Keyword Args
 
 * `allow_inf`: Allow writing of `Inf` and `NaN` values (not part of the JSON standard). [default `false`]
+* `inf_mapping`: A function to map `Inf`, `-Inf` and `NaN` values to a custom representation. [default `nothing`]
+
+    if `inf_mapping` is `nothing` the mapping is equivalent to
+
+    `inf_mapping = x -> x == Inf ? "Infinity" : x == -Inf ? "-Infinity" : "NaN"``
```

Comment:
could we add a comment here about how `inf_mapping` relates to `allow_inf`? does the user _also_ need to specify `allow_inf=true`? or does providing `inf_mapping` automatically imply `allow_inf`?

Reply:
Committed a new version, please review.